### PR TITLE
[FIX] l10n_ec: Fix tax repartition line report expression values

### PR DESCRIPTION
**Pull request by stevTresCloud**
Sales profit tax witholds should not affect the tax report 103, they should be empty

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_ec/migrations/3.5/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    tax_group_id = env.ref('l10n_ec.tax_group_withhold_income_sale')
+    repartition_line_ids = env['account.tax.repartition.line'].search([('invoice_tax_id.tax_group_id', '=', tax_group_id.id)])
+    repartition_line_ids.write({
+        'plus_report_expression_ids': False,
+        'minus_report_expression_ids': False,
+    })
+
+def migrate(cr, version):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    update_withhold_income_sale_repartition_lines(env)
```
Comment (pepetreshere) [inline addons/l10n_ec/migrations/3.5/post-migration.py:16]: @stevTresCloud it is missing to migrate the account.move.lines

**Review by pepetreshere — CHANGES_REQUESTED** [review]

File: addons/l10n_ec/migrations/3.5/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+from odoo import api, SUPERUSER_ID
+
+def update_withhold_income_sale_repartition_lines(env):
+    # Delete tag_ids for withhold income sale repartition lines
+    tax_group_id = env.ref('l10n_ec.tax_group_withhold_income_sale')
+    repartition_line_ids = env['account.tax.repartition.line'].search([('invoice_tax_id.tax_group_id', '=', tax_group_id.id)])
+    repartition_line_ids.write({
+        'plus_report_expression_ids': False,
+        'minus_report_expression_ids': False,
```
Comment (pepetreshere) [inline addons/l10n_ec/migrations/3.5/post-migration.py:11]: @stevTresCloud let's identify the exact tag first, you never know what people might have done

**Review by pepetreshere — CHANGES_REQUESTED** [review]

**Review by pepetreshere — CHANGES_REQUESTED** [review]

File: addons/l10n_ec/migrations/3.5/post-migration.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        account_account_tags_names.append(f'-{tax_code} (Reporte 103)')
+    account_account_tags = env['account.account.tag'].search([('name', 'in', account_account_tags_names)])
+    tax_group_id = env.ref('l10n_ec.tax_group_withhold_income_sale')
+    domain_repartition_line = ['|', ('invoice_tax_id.tax_group_id', '=', tax_group_id.id), ('refund_tax_id.tax_group_id', '=', tax_group_id.id)]
+    repartition_line_ids = env['account.tax.repartition.line'].search(domain_repartition_line)
+    account_move_line_ids = env['account.move.line'].search([('tax_line_id.tax_group_id', '=', tax_group_id.id)])
+    query = '''
+    DELETE FROM account_account_tag_account_tax_repartition_line_rel 
+           WHERE account_tax_repartition_line_id IN {repartition_line_ids}
+           AND account_account_tag_id IN {account_account_tags_ids};
```
Comment (jco-odoo) [inline addons/l10n_ec/migrations/3.5/post-migration.py:21]: with the IN, the entire idea of using SQL gets lost.

Comment (Pirols) [reply addons/l10n_ec/migrations/3.5/post-migration.py:21]: I think that using `IN` is fine in general, we do that all the time in migration scripts ([example](https://github.com/odoo/upgrade/pull/4495/files#diff-def617aa3a377c2bcacdf82e543ab4317c24aa381a21a13c71228c1a9b129289R84-R87)). what we want to improve is the way values are inserted into the queries: do not use `format`/`f-strings`/similars as they may introduce an injection vulnerability in the code. Let, instead the [cursor](https://www.psycopg.org/docs/cursor.html#cursor.execute) do its job by deciding when to put the variables in. Small example:

```
ids = [1,2,3]

cr.execute("DELETE FROM table_name WHERE id IN %s", [tuple(ids)])
```

File: addons/l10n_ec/migrations/3.5/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+def update_withhold_income_sale_repartition_lines(env):
+    # Delete tag_ids for withhold income sale repartition lines
+    # Delete tag_ids for withhold income sale account move lines
+    report_tax_codes = ['343', '393', '346', '396', '344', '394', '3440', '3940', '345', '395']
+    account_account_tags_names = []
+    for tax_code in report_tax_codes:
+        account_account_tags_names.append(f'+{tax_code} (Reporte 103)')
+        account_account_tags_names.append(f'-{tax_code} (Reporte 103)')
+    account_account_tags = env['account.account.tag'].search([('name', 'in', account_account_tags_names)])
```
[the remark this answers is not in the document]
Comment (pepetreshere) [reply addons/l10n_ec/migrations/3.5/post-migration.py]: Finally it resulted that the table doesn't have XML_IDs, neither is linked to tax report lines. So we have only improved the SQL to a more traditional approach

File: addons/l10n_ec/migrations/3.5/post-migration.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, SUPERUSER_ID
+
+def update_withhold_income_sale_repartition_lines(env):
```
Comment (vbe-odoo) [inline addons/l10n_ec/migrations/3.5/post-migration.py:5]: Leave the migration script outside now.
So we can merge it and then we work on the migration script.

File: addons/l10n_ec/__manifest__.py [hunk 6]
```diff
@@ -2,7 +2,7 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 {
     "name": "Ecuadorian Accounting",
-    "version": "3.4",
+    "version": "3.5",
```
Comment (vbe-odoo) [inline addons/l10n_ec/__manifest__.py:5]: Go back to 3.4

**Comment by vbe-odoo** [comment]
@jco-odoo ready for merge.

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**